src/libsyntax/parse/lexer/mod.rs

```diff
@@ -159,11 +165,30 @@ impl<'a> StringReader<'a> {
         self.ch.is_none()
     }
 
-    fn fail_unterminated_raw_string(&self, pos: BytePos, hash_count: u16) {
-        let mut err = self.struct_span_fatal(pos, pos, "unterminated raw string");
-        err.span_label(self.mk_sp(pos, pos), "unterminated raw string");
+    fn fail_unterminated_raw_string(&self, start: Span, hash_count: u16, spans: Vec<Span>) -> ! {
+        const SPAN_THRESHOLD: usize = 3;
+        const MSG_STR: &str = "Raw string could be meant to end here";
+        let hash_str = format!("\"{}", "#".repeat(hash_count as usize));
+        let spans_len = spans.len();
+
+        let mut err = self.sess.span_diagnostic.struct_span_fatal(start, "unterminated raw string");
+        err.span_label(start, "unterminated raw string");
+
+        for s in spans {
+            if spans_len < SPAN_THRESHOLD {
+                err.span_suggestion(s,
+                                    MSG_STR,
+                                    hash_str.clone(),
+                                    Applicability::MaybeIncorrect);
+            } else {
+                err.tool_only_span_suggestion(s,
+                                              MSG_STR,
+                                              hash_str.clone(),
+                                              Applicability::MaybeIncorrect);
```

Comment:
The current style guide is 4 space indent instead of indented on opening delimiter. The codebase hasn't been fully moved to it but we want to avoid adding more code following the old style as possible. In this case it seems like it might even fit as a single line with all the arguments in their own.

Reply:
They don't fit, that's why I did this. Longer than 100 chars and then tidy complains. 
What do you mean with *4 space indent*? Can you give me a short example please (or a link to some code)?

Reply:
```
	                err.tool_only_span_suggestion(
                        s,
                        MSG_STR,
                        hash_str.clone(),
                        Applicability::MaybeIncorrect,
                    );
```